django/db/models/expressions.py

```diff
@@ -483,7 +485,14 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        return '%s', [self.value]
+        val = self.value
+        if self._output_field_or_none is not None:
+            from django.db.models.sql.compiler import SQLUpdateCompiler
+            if isinstance(compiler, SQLUpdateCompiler):
```

Comment:
I haven't had time to test this myself (I will do so tonight), but can we just call `self.value = self.output_field.get_db_prep_value(self.value)` from the Value.resolve_ref method, no matter if the value is being saved or not? I'm not sure I understand the differences between get_db_save and get_db_value though. It seems weird that there are both methods anyway.

Preparing the value for use in SQL shouldn't be different between SELECT and UPDATE, but there may be cases that I'm unaware of that do require differentiation.